LocalFolder/explainer.md

```diff
@@ -0,0 +1,204 @@
+
+We considered creating a new API on `navigator` to return a `FileSystemDirectoryHandle` of the app's LocalFolder directory. This was rejected primarily because of the long term cost of maintaining a non-standard API. 
+
+### File migration
+
+We considered a solution that migrates the contents of the LocalFolder directory into OPFS storage. The drawbacks to this solution includes:
+
+* It could take a long time to migrate large files. There is no way for the calling code to register a callback to wait for migration to be completed without introducing a new API, negating the advantage of not creating a new API on `navigator`.
+* Migration of files could fail due the lack of storage space and other I/O errors. There is no good way to communicate such errors to the calling code without introducing a new API.
+
+## Security risks 
+
+1. A malicious party could attempt to gain access to files outside of the boundary of the `LocalFolder` directory.
+
+1. A malicious party could create executable files on the file system.
+
+1. Third party scripts could read data in `LocalFolder` even though the app developer is not aware of this feature. 
+
+1. Scripts in embedded iframes could try to access data in `LocalFolder`. 
+
+### Mitigations
+
+1. The `FileSystemHandle` design does not support walking up the directory structure.
+
+1. The `LocalFolder` directory handle and its contents will not allow the modification of existing files or creation of new files. This prevents the creation of executable files and the creation of files outside of the LocalFolder directory. Additionally, the implementation of the File System Access API in Chromium prevents creation of some executable file types.
+
+1. The app has to opt in explictly using the `related_applications` field in the web app manifest. The app developer must understand the risks involved and be responsible for the actions of third party scripts used. Most apps that do not intend to access `LocalFolder` will not expose it unknowingly to third parties.
+
+1. The `LocalFolder` handle entry will not be visible to iframes as OPFS storage is partitioned by origin.
+
+
+## Privacy risks
+
+1. `LocalFolder` data in the file system can be accessed by a PWA through a Web API without prompting the user for permission. 
+
+1. Third party scripts can use the presence of `LocalFolder` data to determine if that particular Windows Store app is installed on the client machine. This information could potentially be used for fingerprinting. 
+    * Inspecting `display-mode` is an existing method to check if a site is installed locally as an app but this method does not work when the user navigates the site in a normal browesr tab.  
+    * The fingerprinting surface does not increase significantly as there is any existing method to determine if apps are installed. 
+    * As the installed app information would be different for each site, it cannot be easily used in a comparable fingerprint from different sites. 
```

Comment:
What I'm trying to express: 
You can determine from `a.com` that PFN=A is installed. You can determine from `b.com` that PFN=B is installed. You cannot know from `b.com` that PFN=A is installed. If you use this bit of information in a fingerprint, the fingerprint calculated on a.com is not comparable to the fingerprint calculated on b.com. It can be used to distinguish 2 clients (one installed, one not) on a.com, but it cannot easily be used to track the same client across 2 sites.

Reply:
I'd probably rephrase this.  It's true that you can't determine `a.com`'s installation status from `b.com`, but you can now determine "is _this_ site installed from Microsoft Store".  That additional information could be combined with other existing data that might be able to link the user across both sites (possibly a cookie) and then know which sites have been installed from Microsoft Store.

So, maybe something like this:

` * An origin is only able to use this feature to determine if it itself is installed from the Microsoft Store.  It is unable to determine the installation status of any other origin.  This information can be combined with other unrelated, existing data however to potentially increase the fingerprinting risk.`

Or something like that.

As an aside, you should update all references of `Windows Store` -> `Microsoft Store`.